File: projects/code_combat/7_Sarven_Desert/344-Hot_Gate/hotGate.js
// Use voice commands to command the artillery.

while(true) {
    var enemy = pet.findNearestEnemy();
    if(!enemy) {
        continue;
    }
    // Scouts are fast. We need stop them. 
    if(enemy.type == "scout") {
        var distance = pet.distanceTo(enemy);
        if(pet.isReady("cold-blast") &&  distance < 5) {
            pet.coldBlast();
        }
    }
    else {
        // If the enemy on the left of the pet:
        if (enemy.pos.x < pet.pos.x) {
            // Say  "left".
            pet.say("Left");
        }
        // If the enemy on the right of the pet:
        else {
            // Say  "right".
            pet.say("Right");
        }
    }
}

File: projects/code_combat/7_Sarven_Desert/344-Hot_Gate/hot_gate.py
while True:
    enemy = pet.findNearestEnemy()
    if not enemy:
        continue
    
    if enemy.type == "scout":
        if pet.isReady("cold-blast") and pet.distanceTo(enemy) < 5:
            pet.coldBlast()
    else:
        if enemy.pos.x < pet.pos.x:
            pet.say("Left")
        else:
            pet.say("Right")
